app/src/unitTests/java/com/example/features/dashboard/presenter/MainPresenterTest.java

```diff
@@ -49,7 +53,7 @@
     @Before
     public void setUp() throws Exception {
         MockitoAnnotations.initMocks(this);
-        RxSchedulers rxSchedulers = new RxSchedulers(Schedulers.immediate(), Schedulers.immediate(), Schedulers.immediate(), Schedulers.immediate());
+        RxSchedulers rxSchedulers = new RxSchedulers(Schedulers.io(), Schedulers.computation(), Schedulers.trampoline(), AndroidSchedulers.mainThread());
         mainPresenter = new MainPresenter(mockRestService, rxSchedulers, mockAnalyticsHelper, new ShotMapper(), new CountingIdlingResource("test"));
         mainPresenter.attachView(mockMainView);
     }
```

Comment:
there's another issue with this class: One of the tests is failing. Test with name:
"null_shots_are_filtered_before_returned"

This is because in the second line of the test there is a 
```
        shotResponseList.add(null);
```

The test seems to be failing because in the MainPresenter 
```
Observable::fromIterable
```
has a null check, i.e. in RxJava 2 null values are no longer allowed (i think)

So to sum up:


a)
in the MainPresenter please change line

```
                .filter(shot -> shot != null && shot.getImagesData() != null && shot.getImagesData().getTeaserImageUrl() != null)

```

to 
```
                .filter(shot -> shot.getImagesData() != null && shot.getImagesData().getTeaserImageUrl() != null)
```

in this test "null_shots_are_filtered_before_returned"

please delete the line 

```
        shotResponseList.add(null);
```